config.go

```diff
@@ -549,6 +551,54 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
+type githubAppAuth struct {
+	AppId      int64
+	privateKey []byte
+}
+
+// getGithubAppAuth returns app id and app private key if it exists.
+func (s *Settings) getGithubAppAuth() *githubAppAuth {
+	if s.AuthConfig.Github == nil || s.AuthConfig.Github.AppId == 0 {
+		return nil
+	}
+
+	key := s.Expansions[githubAppPrivateKey]
+	if key == "" {
+		return nil
+	}
+
+	return &githubAppAuth{
+		AppId:      s.AuthConfig.Github.AppId,
+		privateKey: []byte(key),
+	}
+}
+
+// CreateInstallationToken uses the owner/repo information to request an github app installation id
+// and uses that id to create an installation token.
+func (s *Settings) CreateInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+	authFields := s.getGithubAppAuth()
+	if authFields == nil {
+		return "", errors.New("Github app settings are not set")
+	}
+	httpClient := utility.GetHTTPClient()
+	itr, err := ghinstallation.NewAppsTransport(httpClient.Transport, authFields.AppId, authFields.privateKey)
+	if err != nil {
+		return "", errors.Wrap(err, "creating transport with JWT")
+	}
+	httpClient.Transport = itr
+	defer utility.PutHTTPClient(httpClient)
+	client := github.NewClient(httpClient)
+	installationId, _, err := client.Apps.FindRepositoryInstallation(ctx, owner, repo)
+	if err != nil || installationId == nil {
+		return "", errors.Wrapf(err, "finding installation token for '%s/%s'", owner, repo)
```

Comment:
This needs to make a new error in the case where `err` is non-nil but installationId is nil (or make a separate if block).